src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/TraceUtilities.cs

```diff
@@ -97,5 +99,49 @@ public static async Task CaptureTraceAsync(TaskCompletionSource<object> startCom
 
             await pipeProcessor.RunAsync(token);
         }
+
+        public static async Task CaptureTraceUntilEventAsync(TaskCompletionSource<object> startCompletionSource, IEndpointInfo endpointInfo, MonitoringSourceConfiguration configuration, TimeSpan timeout, Stream outputStream, string providerName, string eventName, IDictionary<string, string> payloadFilter, ILogger logger, CancellationToken token)
+        {
+            DiagnosticsClient client = new(endpointInfo.Endpoint);
+            TaskCompletionSource<object> stoppingEventHitSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            using IDisposable registration = token.Register(
+                () => stoppingEventHitSource.TrySetCanceled(token));
+
+            await using EventTracePipeline pipeProcessor = new(client, new EventTracePipelineSettings
+            {
+                Configuration = configuration,
+                Duration = timeout,
+            },
+            async (eventStream, token) =>
+            {
+                startCompletionSource?.TrySetResult(null);
```

Comment:
This is just a thought, not something we need to implement now, but it's very likely that the delegate code to create a new EventMonitoringPassThroughStream, pass the respective parameters, and stop tracing based on an event is going to be very boilerplate. I'm wondering if we can encapsulate that code as something the user can just pass directly into the EventTracePipeline rather than having each user re-write the boilerplate code.